src/material/schematics/ng-generate/m3-theme/index.ts

```diff
@@ -182,142 +338,114 @@ function createThemeFile(scss: string, tree: Tree, directory?: string) {
       colorComment += ', secondary: ' + options.secondaryColor;
+      secondaryPalette = TonalPalette.fromHct(getHctFromHex(options.secondaryColor));
+    } else {
+      secondaryPalette = TonalPalette.fromHueAndChroma(
+        primaryColorHct.hue,
+        Math.max(primaryColorHct.chroma - 32.0, primaryColorHct.chroma * 0.5),
+      );
     }
+
+    let tertiaryPalette;
     if (options.tertiaryColor) {
-      colorPalettes.set('tertiary', getColorTonalPalettes(options.tertiaryColor).get('primary')!);
       colorComment += ', tertiary: ' + options.tertiaryColor;
+      tertiaryPalette = TonalPalette.fromHct(getHctFromHex(options.tertiaryColor));
+    } else {
+      tertiaryPalette = TonalPalette.fromInt(
+        DislikeAnalyzer.fixIfDisliked(
+          new TemperatureCache(primaryColorHct).analogous(3, 6)[2],
+        ).toInt(),
+      );
     }
+
+    let neutralPalette;
     if (options.neutralColor) {
-      colorPalettes.set('neutral', getColorTonalPalettes(options.neutralColor).get('primary')!);
       colorComment += ', neutral: ' + options.neutralColor;
+      neutralPalette = TonalPalette.fromHct(getHctFromHex(options.neutralColor));
+    } else {
+      neutralPalette = TonalPalette.fromHueAndChroma(
+        primaryColorHct.hue,
+        primaryColorHct.chroma / 8.0,
+      );
     }
 
     if (!options.themeTypes) {
       context.logger.info('No theme types specified, creating both light and dark themes.');
       options.themeTypes = 'both';
     }
 
-    const themeScss = generateSCSSTheme(
-      colorPalettes,
-      options.themeTypes,
+    if (options.generateHighContrastOverrideMixins && !options.useSystemVariables) {
+      context.logger.info(
+        'High contrast theme mixins require using system variables to work. Enabling system variables for generated theme.',
+      );
+      options.useSystemVariables = true;
+    }
+
+    // Generate standard contrast theme
+    let themeTypes = options.themeTypes === 'both' ? ['light', 'dark'] : [options.themeTypes];
+    const scheme = getMaterialDynamicScheme(
+      primaryPalette,
+      secondaryPalette,
+      tertiaryPalette,
+      neutralPalette,
+      neutralVariantPalette,
+      false,
+      0,
+    );
+    let themeScss = generateSCSSTheme(
+      getColorPalettesMapFromDynamicScheme(scheme),
+      themeTypes,
       colorComment,
       options.useSystemVariables || false,
+      options.systemVariablePrefix,
     );
-    createThemeFile(themeScss, tree, options.directory);
-  };
-}
-
-/**
- * The hue map produced by `material-color-utilities` may miss some neutral hues depending on
- * the provided colors. This function estimates the missing hues based on the generated ones
- * to ensure that we always produce a full palette. See #29157.
- *
- * This is a TypeScript port of the logic in `core/theming/_palettes.scss#_patch-missing-hues`.
- */
-function patchMissingHues(
```

Comment:
How are we accounting for these missing hues now or did the module get fixed so it generates everything now? If that's the case, we may want to update `src/material/core/theming/_palettes.scss` which was doing similar patching.

Reply:
Things get generated correctly now, it has those missing tones that we usually add from this patching when the theme gets generated.

I can update that file with the missing values in a separate PR so we don't have to patch anymore for our predefined palettes.